### ISSUE TITLE
Annotate remaining runtime libraries for NativeAOT

### ISSUE THREAD

**Issue by LakshanF** [dotnet/runtime#75480]
The below set of runtime assemblies need to be annotated for AOT (the # of warnings might not be accurate and need to be validated). Related issues - #69739, #61231, #67741

Assembly | Aot Warnings
-- | --
System.Linq.Expressions.dll | 124
System.Private.Xml.dll | 78
System.Xml.XmlSerializer.dll | 39
System.Xml.XmlDocument.dll | 39
System.Xml.XPath.dll | 39
System.Xml.XPath.XDocument.dll | 39
System.Private.Xml.Linq.dll | 39
System.Xml.ReaderWriter.dll | 39
System.Xml.XDocument.dll | 39
Microsoft.CSharp.dll | 20
System.ComponentModel.Composition | ?
System.Composition.Hosting | ?
System.Composition.Runtime | ?
System.Composition.TypedParts | ?
System.Configuration.ConfigurationManager | ?
System.Data.Odbc | ?
System.Data.OleDb | ?
System.DirectoryServices | ?
System.DirectoryServices.AccountManagement | ?
System.DirectoryServices.Protocols | ?
System.Drawing.Common | ?
System.Reflection.Context | ?
System.Reflection.MetadataLoadContext | ?
System.Resources.Extensions | ?
System.Runtime.InteropServices.JavaScript | ?
System.Runtime.Serialization.Schema | ?
System.ServiceModel.Syndication | ?
System.Windows.Extensions | ?

**Comment by danmoseley** [comment]
our systems expect exactly 1 area label -- I'm assuming you're doing it @LakshanF so leaving area-nativeaot-coreclr.

**Comment by JamesNK** [comment]
`System.Linq.Expressions` warnings appear when publishing an empty ASP.NET Core app.

**Comment by jkotas** [comment]
What's the warning? Should `System.Linq.Expressions` be trimmed out from ASP.NET Core apps in .NET 8 instead?

**Comment by JamesNK** [comment]
> What's the warning?

```
/_/src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/ArrayOperations.cs(23): AOT analysis warning IL3050: System.Linq.Expressions.Interpreter.NewArrayInitInstruction.Run(InterpretedFrame): Using member 'System.Array.CreateInstance(Type,Int32)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. The code for an array of the specified type might not be available. [C:\DotNetTest\AspNetTest\AspNetTest.csproj]
/_/src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/ArrayOperations.cs(52): AOT analysis warning IL3050: System.Linq.Expressions.Interpreter.NewArrayInstruction.Run(InterpretedFrame): Using member 'System.Array.CreateInstance(Type,Int32)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. The code for an array of the specified type might not be available. [C:\DotNetTest\AspNetTest\AspNetTest.csproj]
/_/src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/ArrayOperations.cs(87): AOT analysis warning IL3050: System.Linq.Expressions.Interpreter.NewArrayBoundsInstruction.Run(InterpretedFrame): Using member 'System.Array.CreateInstance(Type,Int32[])' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. The code for an array of the specified type might not be available. [C:\DotNetTest\AspNetTest\AspNetTest.csproj]
/_/src/libraries/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs(28): AOT analysis warning IL3050: System.Dynamic.Utils.TypeUtils.GetNullableType(Type): Using member 'System.Type.MakeGenericType(Type[])' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. The native code for this instantiation might not be available at runtime. [C:\DotNetTest\AspNetTest\AspNetTest.csproj]
/_/src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/MethodCallExpression.cs(1379): AOT analysis warning IL3050: System.Linq.Expressions.Expression.ApplyTypeArgs(MethodInfo,Type[]): Using member 'System.Reflection.MethodInfo.MakeGenericMethod(Type[])' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. The native code for this instantiation might not be available at runtime. [C:\DotNetTest\AspNetTest\AspNetTest.csproj]
/_/src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/BinaryExpression.cs(2239): AOT analysis warning IL3050: System.Linq.Expressions.Expression.GetResultTypeOfShift(Type,Type): Using member 'System.Type.MakeGenericType(Type[])' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. The native code for this instantiation might not be available at runtime. [C:\DotNetTest\AspNetTest\AspNetTest.csproj]
```

> Should System.Linq.Expressions be trimmed out from ASP.NET Core apps in .NET 8 instead?

I think that is a goal. There are some places where `System.Linq.Expressions` is coupled to important code. I'll investigate more whether there are ways to completely remove any code that touches on it.

**Comment by eerhardt** [comment]
> Should System.Linq.Expressions be trimmed out from ASP.NET Core apps in .NET 8 instead?

Yes, my intention is that Linq.Expressions won't be used in basic ASP.NET Core app in .NET 8. From my previous investigations, it adds ~1MB to the app size.

With https://github.com/dotnet/aspnetcore/pull/46020#issuecomment-1380045771 we have eliminated it from the core logic. And saved 1.3 MB.

**Comment by LakshanF** [comment]
This should be done now.

**Comment by eerhardt** [comment]
This is not done. We still disable the AOT analyzer in the following projects:

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.Linq.Expressions/src/System.Linq.Expressions.csproj#L6

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.Private.Xml/src/System.Private.Xml.csproj#L6

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/Microsoft.CSharp/src/Microsoft.CSharp.csproj#L13

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.DirectoryServices.AccountManagement/src/System.DirectoryServices.AccountManagement.csproj#L9

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.DirectoryServices/src/System.DirectoryServices.csproj#L7

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.ServiceModel.Syndication/src/System.ServiceModel.Syndication.csproj#L6

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.Resources.Extensions/src/System.Resources.Extensions.csproj#L6

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.DirectoryServices.Protocols/src/System.DirectoryServices.Protocols.csproj#L13

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.Configuration.ConfigurationManager/src/System.Configuration.ConfigurationManager.csproj#L8

We need to annotate those libraries for NativeAOT and enable the analyzer.
https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.Reflection.Context/src/System.Reflection.Context.csproj#L4

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj#L9

https://github.com/dotnet/runtime/blob/efde0061d4fadce140d9c9bd78bfc2aa7df7d9eb/src/libraries/System.Reflection.MetadataLoadContext/src/System.Reflection.MetadataLoadContext.csproj#L6
